src/hobbits-plugins/importerexporters/UsbDevice/usbdevice.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "importexportinterface.h"
+#include "parameterdelegate.h"
+#include <libusb-1.0/libusb.h>
+
+class UsbDevice : public QObject, ImporterExporterInterface
+{
+    Q_OBJECT
+    Q_PLUGIN_METADATA(IID "hobbits.ImporterExporterInterface.UsbDevice")
+    Q_INTERFACES(ImporterExporterInterface)
+
+public:
+    UsbDevice();
+
+    ImporterExporterInterface* createDefaultImporterExporter() override;
+
+    QString name() override;
+    QString description() override;
+    QStringList tags() override;
+
+    bool canExport() override;
+    bool canImport() override;
+
+    virtual QSharedPointer<ParameterDelegate> importParameterDelegate() override;
+    virtual QSharedPointer<ParameterDelegate> exportParameterDelegate() override;
+
+    QSharedPointer<ImportResult> importBits(const Parameters &parameters,
+                                            QSharedPointer<PluginActionProgress> progress) override;
+
+
+    QSharedPointer<ExportResult> exportBits(QSharedPointer<const BitContainer> container,
+                                            const Parameters &parameters,
+                                            QSharedPointer<PluginActionProgress> progress) override;
+
+   struct usbParams setupLibusb(usbParams params);
+    QSharedPointer<ImportResult> returnError(int errorCode);
+   struct usbParams exitLibusb(bool closeDevice, usbParams params);
+
+private:
+    QSharedPointer<ParameterDelegate> m_importDelegate;
+    QSharedPointer<ParameterDelegate> m_exportDelegate;
+   
+};
+
+struct usbParams
```

Comment:
use the same type naming syntax that we are using for classes: `UsbParams`

starting with lowercase implies that it is a variable, not a type

Reply:
also, I think you want to typedef the struct name https://stackoverflow.com/questions/1675351/typedef-struct-vs-struct-definitions